integrations/crewai/README.md

```diff
@@ -0,0 +1,211 @@
+)
+
+# sets the default uc function client
+set_uc_function_client(client)
+```
+
+#### Create a UC function
+
+To provide an executable function for your tool to use, you need to define and create the function within UC. To do this,
+create a Python function that is wrapped within the SQL body format for UC and then utilize the `DatabricksFunctionClient` to store this in UC:
+
+```python
+# Replace with your own catalog and schema for where your function will be stored
+CATALOG = "catalog"
+SCHEMA = "schema"
+function_name = f"{CATALOG}.{SCHEMA}.make_uppercase"
+
+def make_uppercase(s: str) -> str:
+    """
+    Convert the string to uppercase.
+    """
+    return s.upper()
+
+response = client.create_python_function(func=make_uppercase, catalog=CATALOG, schema=SCHEMA)
+```
+
+Now that the function exists within the Catalog and Schema that we defined, we can interface with it from CrewAI using the ucai_crewai package.
+
+#### Create an instance of a CrewAI compatible tool
+
+[CrewAI Tools](https://docs.crewai.com/core-concepts/Tools/) are callable external functions that GenAI applications can use (called by
+an LLM), which are exposed with a UC interface through the use of the ucai_crewai package via the `UCFunctionToolkit` API.
+
+```python
+from ucai_crewai.toolkit import UCFunctionToolkit
+
+# Pass the UC function name that we created to the constructor
+toolkit = UCFunctionToolkit(function_names=[function_name])
+
+# Get the CrewAI-compatible tools definitions
+tools = toolkit.tools
+```
+
+If you would like to validate that your tool is functional prior to integrating it with CrewAI, you can call the tool directly:
+
+```python
+my_tool = tools[0]
+
+my_tool.fn(**{"s": "lowercase string"})
+```
+
+#### Utilize our function as a tool within a CrewAI `Crew`
+
+With our interface to our UC function defined as a CrewAI tool collection, we can directly use it within a CrewAI `Crew`. 
+
+```python
+import os
+from crewai import Agent, Task, Crew
+
+# Set up API keys
+os.environ["OPENAI_API_KEY"] = "your key"
+
+# Create agents
+coder = Agent(
+    role="Simple coder",
+    goal= "Create a program that prints Hello Unity Catalog!",
+    backstory="likes long walks on the beach",
+    expected_output="string",
+    tools=tools,
+    verbose=True
+)
+
+reviewer = Agent(
+    role="reviewer",
+    goal="Ensure the researcher calls a function and shows the answer",
+    backstory="allergic to cats",
+    expected_output="string",
+    verbose=True
+)
+
+# Define tasks
+research = Task(
+    description="Call a tool",
+    expected_output="string",
+    agent=coder
+)
+
+review = Task(
+    description="Review the tool call output. Once complete, stop.",
+    expected_output="string",
+    agent=reviewer,
+)
+
+# Assemble a crew with planning enabled
+crew = Crew(
+    agents=[coder, reviewer],
+    tasks=[research, review],
+    verbose=True,
+    planning=True,  # Enable planning feature
+)
+
+# Execute tasks
+crew.kickoff()
```

Comment:
most hilariously named method I've ever seen in a major OSS package.

Reply:
+1